work_queue/src/work_queue_json.c

```diff
@@ -0,0 +1,341 @@
+    if(command_line){
+
+        struct work_queue_task* task = work_queue_task_create(command_line);
+
+        if (!task){
+            return NULL;
+        }
+
+        if(input_files){            
+            specify_files(1, input_files, task);
+        }
+
+        if(output_files){
+            specify_files(0, output_files, task);
+        }
+    
+        return task;
+
+    }
+
+    return NULL;
+
+}
+
+/** Create a new work_queue object.
+@param port The port number to listen on. If zero is specified, then the 
+port stored in <b>WORK_QUEUE_PORT</b> is used if available. If it isn't, or 
+-1 is specified, the first unused port between <b>WORK_QUEUE_LOW_PORT</b> 
+and <b>WORK_QUEUE_HIGH_PORT</b> (1024 and 32767 by default) is chosen.
+@return A new work queue, or null if it could not be created. 
+ */
+struct work_queue* work_queue_json_create(const char* str){
+
+
+    int port=0, priority=0;
+    char *name;
+
+    struct jx* json = jx_parse_string(str);
+    if(!json){
+        return NULL;
+    }
+
+    //validate json
+    if(validate_json(json, work_queue_properties)){
+        return NULL;
+    }
+
+    void *i = NULL;
+    void *j = NULL;
+    const char* key = jx_iterate_keys(json, &j);
+    struct jx* value = jx_iterate_values(json, &i);
+
+    while (key != NULL) {
+
+        if (!strcmp(key, "name")){
+            name = value->u.string_value;
+        }
+        else if (!strcmp(key, "port")){
+            port = value->u.integer_value;
+        }
+        else if (!strcmp(key, "priority")){
+            priority = value->u.integer_value;
+        }
+        else{
+            printf("Not necessary: %s\n", value->u.string_value);
+        }
+
+        key = jx_iterate_keys(json, &j);
+        value = jx_iterate_values(json, &i);
+
+    }    
+
+    if(port){
+
+        struct work_queue *workqueue = work_queue_create(port);
+
+        if(!workqueue){ 
+            return NULL; 
+        }        
+
+        if(name){
+            work_queue_specify_name(workqueue, name);
+        }
+        if(priority){
+            work_queue_specify_priority(workqueue, priority);
+        }
+    
+        return workqueue;
+
+    }
+
+    return NULL;
+
+}
+
+/** Submit a task to a queue.
+Once a task is submitted to a queue, it is not longer under the user's 
+control and should not be inspected until returned via @ref work_queue_wait.
+Once returned, it is safe to re-submit the same take object via 
+@ref work_queue_submit.
+@param q A work queue object.
+@param str A JSON description of a task. 
+
+{ "command_line" : "<i>string</i>" , "output_files" : <i>array of objects with one object per output file</i> -> 
+[ { "local" : "<i>string</i>" , "remote" : "<i>string</i>" , "flags" : "<i>integer</i>" } ] , "input _files" : 
+<i>array of objects with one object per input file</i> -> [ { "local" : "<i>string</i>" , "remote" : 
+"<i>string</i>" , "flags" : "<i>integer</i>" } ] , "tag" : "<i>string</i>" }
+
```

Comment:
This is a good start at a task representation.

Have a look at the JSON representation of a task in the JX workflow language, and try to synchronize up the field names, as much as possible.

I'm a little uncomfortable representing the `flags` as an integer, since that's hard to create/understand in a JSON context.  Is there a more JSON native way of expressing a set of options for each file?